designs/ble-mote.stanza

```diff
@@ -72,7 +72,7 @@ pcb-module sensor-mote :
   place(ant) at loc(9.0, 5.0, -90.0) on Top
 
   ; Use a connect statement to interface USB-2 to UART
-  connect-ft232(protected-usb, proc.uart)
+  val cp2105 = connect-cp2105(protected-usb, proc.uart) as JITXObject
```

Comment:
why is this cast needed?

Reply:
I wonder as well.

Reply:
Me too! I can't net to the ports of the returned module without the cast.

Reply:
It's due to the definition of `connect-cp2105` which may fail:

```
public defn connect-cp2105 (usb:JITXObject, uart:JITXObject) :
  inside pcb-module :
    match(port-type(usb), port-type(uart)) :
      (px:Bundle, py:Bundle) :
        if px == usb-2 and py == uart-with([`dtr `rts]) :
          public inst xcvr : ocdb/si-labs/CP2105/module
          net (usb, xcvr.usb-2)
          net (uart.tx, xcvr.e-uart.rx)
          net (uart.rx, xcvr.e-uart.tx)
          net (uart.dtr, xcvr.e-uart.dtr)
          net (uart.rts, xcvr.e-uart.rts)
          xcvr
```

Which returns `False|JITXObject`. We should refactor all the `connects-xxx (...)` in connects.stanza to check their preconditions rather than rely on API consumers to do the casting.

Reply:
Agreed. Let's put the fatal inside the `connect-cp2105` function.

Reply:
I added the checks in #56